sycl/test/fpga_tests/fpga_pipes.cpp

```diff
@@ -0,0 +1,180 @@
+      auto write_acc = writeBuf.get_access<access::mode::write>(cgh);
+
+      cgh.single_task<class goo_nb>([=]() {
+        bool SuccessCode = false;
+        while (!SuccessCode)
+          write_acc[0] = Pipe::read(SuccessCode);
+      });
+    });
+
+    auto readHostBuffer = writeBuf.get_access<access::mode::read>();
+    if (readHostBuffer[0] != 42) {
+      std::cout << "Result mismatches " << readHostBuffer[0] << " Vs expected "
+                << 42 << std::endl;
+
+      return -1;
+    }
+  }
+
+  {
+    // Test for simple non-blocking pipes with explicit type
+    queue Queue;
+
+    buffer<int, 1> readBuf(data, 1);
+    Queue.submit([&](handler &cgh) {
+      cgh.single_task<class boo_nb>([=]() {
+        bool SuccessCode;
+        while (!SuccessCode)
+          pipe<class some_pipe, int, 1>::write(42, SuccessCode);
+      });
+    });
+
+    buffer<int, 1> writeBuf(data, 1);
+    Queue.submit([&](handler &cgh) {
+      auto write_acc = writeBuf.get_access<access::mode::write>(cgh);
+
+      cgh.single_task<class zoo_nb>([=]() {
+        bool SuccessCode;
+        while (!SuccessCode)
+          write_acc[0] = pipe<class some_pipe, int, 1>::read(SuccessCode);
+      });
+    });
+
+    auto readHostBuffer = writeBuf.get_access<access::mode::read>();
+    if (readHostBuffer[0] != 42) {
+      std::cout << "Result mismatches " << readHostBuffer[0] << " Vs expected "
+                << 42 << std::endl;
+
+      return -1;
+    }
+  }
+
+  {
+    // Test for simple non-blocking pipes created with namespaces set
+    queue Queue;
+
+    buffer<int, 1> readBuf(data, 1);
+    Queue.submit([&](handler &cgh) {
+      cgh.single_task<class foo_ns>([=]() {
+        bool SuccessCode;
+        while (!SuccessCode)
+          pipe<class some::pipe, int, 1>::write(42, SuccessCode);
+      });
+    });
+
+    buffer<int, 1> writeBuf(data, 1);
+    Queue.submit([&](handler &cgh) {
+      auto write_acc = writeBuf.get_access<access::mode::write>(cgh);
+
+      cgh.single_task<class boo_ns>([=]() {
+        bool SuccessCode;
+        while (!SuccessCode)
+          write_acc[0] = pipe<class some::pipe, int, 1>::read(SuccessCode);
+      });
+    });
+
+    auto readHostBuffer = writeBuf.get_access<access::mode::read>();
+    if (readHostBuffer[0] != 42) {
+      std::cout << "Result mismatches " << readHostBuffer[0] << " Vs expected "
+                << 42 << std::endl;
+
+      return -1;
+    }
+  }
+
+  {
+    // Test for forward declared pipes
+    queue Queue;
+    class pipe_type_for_lambdas;
+
+    buffer<int, 1> readBuf(data, 1);
+    Queue.submit([&](handler &cgh) {
+      cgh.single_task<class foo_la>([=]() {
+        bool SuccessCode;
+        while (!SuccessCode)
+          pipe<class pipe_type_for_lambdas, int>::write(42, SuccessCode);
+      });
+    });
+
+    buffer<int, 1> writeBuf(data, 1);
+    Queue.submit([&](handler &cgh) {
+      cgh.single_task<class boo_la>([=]() {
+        bool SuccessCode;
+        while (!SuccessCode)
```

Comment:
Half-baked: this is undefined behaviour since it should be a `do { ... } while(!SuccessCode);`